packages/squiggle-lang/__tests__/library/list_test.ts

```diff
@@ -154,5 +214,27 @@ describe("List functions", () => {
   describe("flatten", () => {
     testEvalToBe("List.flatten([[1,2], [3,4]])", "[1,2,3,4]");
     testEvalToBe("List.flatten([[1,2], [3,[4,5]]])", "[1,2,3,[4,5]]");
+    testEvalToBe("List.flatten([])", "[]");
+    testEvalToBe("List.flatten([[],[],[]])", "[]");
+  });
+
+  describe("zip", () => {
+    testEvalToBe("List.zip([1,2], [3,4])", "[[1,3],[2,4]]");
+    testEvalToBe(
+      "List.zip([1,2,4], [3,4])",
+      "Error(Error: List lengths must be equal)"
```

Comment:
TODO: It would be nice if we could have some other error type here. Like ArgumentError.